### ISSUE TITLE
something wrong with the ordering xValues in dose response

### ISSUE THREAD

**Issue by elbaraim** [PEtab-dev/PEtab#414]
Maybe a bit difficult to explain here, but i will try my best :smile: 
I am generating a dose response plot and the some of the values in the condition table are not numerically sorted. This has been previously addressed in #396 
if I naturally sort (numerically) increasingly the `simulationConditionId` without changing the condition file, I get the expected plot:
![Screenshot from 2020-03-25 17-18-17](https://user-images.githubusercontent.com/10628853/77560647-07af6c00-6ebe-11ea-83cb-26c279a140b4.png)

However, I noticed that if I do the same BUT sorting the `simulationConditionId` by, e.g., `model1_data1`,`model1_data10`,`model1_data2`,`model1_data2` .... (not numerically increasing, default sorting) and accordingly the full experimentalData table is also sorted, I obtain this:
![Screenshot from 2020-03-25 17-18-39](https://user-images.githubusercontent.com/10628853/77560634-041be500-6ebe-11ea-8cdf-e5ecc5a8f18f.png)

**Comment by dweindl** [comment]
> Maybe a bit difficult to explain here

Best to provide a (minimal) example reproducing the unexpected behavior :)

**Comment by elbaraim** [comment]
Good point @dweindl, can I just simply refer to the Benchmark-Models-PEtab? or should I upload the corresponding files?

**Comment by dweindl** [comment]
Link to there with info on something like dataset ID should do.

**Comment by elbaraim** [comment]
> Link to there with info on something like dataset ID should do.
> 
> The already sorted file and only the affected plot would be the easiest though.

sounds good, I will prepare that

**Comment by elbaraim** [comment]
So execute the following piece of code:
``` python
from petab.visualize import plot_data_and_simulation
import matplotlib.pyplot as plt
%matplotlib inline

model_name = "Casaletto_PNAS2019"

# These files generate the correct figure
data_file_path = "./measurementData1_"+model_name+".tsv"
condition_file_path = "./experimentalCondition_"+model_name+".tsv"
visualization_file_path = "./visualizationSpecification1_"+model_name+".tsv"
simulation_file_path = "./simulatedData1_"+model_name+".tsv"

ax = plot_data_and_simulation(data_file_path,
                              condition_file_path,
                              visualization_file_path,
                             simulation_file_path)

# These files do not generate the correct figure
data_file_path = "./measurementData2_"+model_name+".tsv"
condition_file_path = "./experimentalCondition_"+model_name+".tsv"
visualization_file_path = "./visualizationSpecification1_"+model_name+".tsv"
simulation_file_path = "./simulatedData2_"+model_name+".tsv"

ax = plot_data_and_simulation(data_file_path,
                              condition_file_path,
                              visualization_file_path,
                             simulation_file_path)
```
Using the following files [here](https://github.com/PEtab-dev/PEtab/files/4385572/files.zip).
You should obtain the following figures (yes the simulated data is a bit off from the data, but this should not affect the problem):
![Screenshot from 2020-03-26 08-18-09](https://user-images.githubusercontent.com/10628853/77620195-1fc4d100-6f3a-11ea-9adc-0edae7f8240b.png)

I hope this helps.

**Comment by yannikschaelte** [comment]
@erikadudki this should be closed via #430. Please reopen if not.
